### PR TITLE
feature: adding smarter unit test execution

### LINKED ISSUE #5965: Smarter Unit Test Execution
Right now all unit tests run during a plan unless the `--skip-tests` flag is used.

We already have the plan which shows us which models are affected by the changes. It makes sense to trim the unit tests that run during a plan to only the models that are affected by applying the plan.

I am not sure what behavior should happen for the `--skip-backfill` / `--dry-run` flags, but my initial thought is to skip all tests for this flag as well since it's considered a no-op operation.

To have backwards compatibility, we can add in an `--all-tests` flag to the plan command. I can see this being useful for CI dry-runs as well as edge cases that may show up.

### PR DESCRIPTION
**Pull request by cmgoffena13**
## Description

- Add in `--test-changed-only` flag to only run tests for changed models in the plan.
- Default behavior remains running all tests for all models
- Add `--select-model` flag to the `test` command for improved functionality

Addresses #5965

## Test Plan
- [X] Plan runs all unit tests by default (including no-change plans)
- [X] Plan with model changes runs the full suite unless --test-changed-only is set
- [X] --test-changed-only runs only changed/restated models’ unit tests
- [X] Plan with no changes and --test-changed-only skips unit tests
- [X] --skip-tests still skips; incompatible with --test-changed-only
- [X] Docs cover run-all default and --test-changed-only

## Checklist

- [X] I have run `make style` and fixed any issues
- [X] I have added tests for my changes (if applicable)
- [X] All existing tests pass (`make fast-test`)
- [X] My commits are signed off (`git commit -s`) per the [DCO](DCO)

### CODE REVIEW

File: sqlmesh/cli/main.py [hunk 1]
```diff
@@ -423,6 +423,12 @@ def diff(ctx: click.Context, environment: t.Optional[str] = None) -> None:
     help="Skip tests prior to generating the plan if they are defined.",
     default=None,
 )
+@click.option(
+    "--all-tests",
```
Comment (mday-io) [inline sqlmesh/cli/main.py]: Most of the other flags are `verb_*`. should this be `--include_all_tests`, or something similar?

**Review by mday-io — APPROVED** [review]
